woocommerce/woocommerce
# utm params for add payment method notice

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




Add UTM parameters to missing payment method notice links.

Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. run `nvm use`, `pnpm install`, and `pnpm build`
2. go to `plugins/woocommerce` and run `pnpm -- wp-env start`
3. apply these following patch to force missing payment method notice to be shown
```
Index: plugins/woocommerce/src/Admin/PluginsHelper.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/plugins/woocommerce/src/Admin/PluginsHelper.php b/plugins/woocommerce/src/Admin/PluginsHelper.php
--- a/plugins/woocommerce/src/Admin/PluginsHelper.php	(revision 8c9a6b8ca16f34912a7c69c53647a1a17daf7e2f)
+++ b/plugins/woocommerce/src/Admin/PluginsHelper.php	(date 1721287690098)
@@ -801,36 +801,36 @@
 		if ( ! WC_Helper::is_site_connected() ) {
 			return array();
 		}
-
-		if ( ! self::$can_show_expiring_subs_notice ) {
-			return array();
-		}
-
-		if ( ! self::should_show_notice( self::DISMISS_EXPIRING_SUBS_NOTICE ) ) {
-			return array();
-		}
-
-		$subscriptions          = WC_Helper::get_subscription_list_data();
-		$expiring_subscriptions = array_filter(
-			$subscriptions,
-			function ( $sub ) {
-				return ( ! empty( $sub['local']['installed'] ) && ! empty( $sub['product_key'] ) )
-						&& $sub['expiring']
-						&& ! $sub['autorenew'];
-			},
-		);
-
-		if ( ! $expiring_subscriptions ) {
-			return array();
-		}
-
-		$total_expiring_subscriptions = count( $expiring_subscriptions );
+//
+//		if ( ! self::$can_show_expiring_subs_notice ) {
+//			return array();
+//		}
+//
+//		if ( ! self::should_show_notice( self::DISMISS_EXPIRING_SUBS_NOTICE ) ) {
+//			return array();
+//		}
+//
+//		$subscriptions          = WC_Helper::get_subscription_list_data();
+//		$expiring_subscriptions = array_filter(
+//			$subscriptions,
+//			function ( $sub ) {
+//				return ( ! empty( $sub['local']['installed'] ) && ! empty( $sub['product_key'] ) )
+//						&& $sub['expiring']
+//						&& ! $sub['autorenew'];
+//			},
+//		);
+//
+//		if ( ! $expiring_subscriptions ) {
+//			return array();
+//		}
+//
+//		$total_expiring_subscriptions = count( $expiring_subscriptions );
 
 		// When payment method is missing on WooCommerce.com.
-		$helper_notices = WC_Helper::get_notices();
-		if ( ! empty( $helper_notices['missing_payment_method_notice'] ) ) {
-			return self::get_missing_payment_method_notice( $allowed_link, $total_expiring_subscriptions );
-		}
+//		$helper_notices = WC_Helper::get_notices();
+//		if ( ! empty( $helper_notices['missing_payment_method_notice'] ) ) {
+			return self::get_missing_payment_method_notice( $allowed_link, 1 );
+//		}
 
 		// Payment method is available but there are expiring subscriptions.
 		$notice_data = self::get_subscriptions_notice_data(
Index: plugins/woocommerce/includes/admin/helper/class-wc-helper.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/plugins/woocommerce/includes/admin/helper/class-wc-helper.php b/plugins/woocommerce/includes/admin/helper/class-wc-helper.php
--- a/plugins/woocommerce/includes/admin/helper/class-wc-helper.php	(revision 8c9a6b8ca16f34912a7c69c53647a1a17daf7e2f)
+++ b/plugins/woocommerce/includes/admin/helper/class-wc-helper.php	(date 1721287712250)
@@ -2181,6 +2181,7 @@
 	 * @return bool
 	 */
 	public static function is_site_connected(): bool {
+		return true;
 		$auth = WC_Helper_Options::get( 'auth' );
 
 		// If `access_token` is empty, there's no active connection.

```
4.  go to the http://localhost:8888/wp-admin/admin.php?page=wc-settings and verify that the link includes the `utm_source` and `utm_campaign` 
<img width="1106" alt="image" src="https://github.com/user-attachments/assets/04a4ae6e-d1af-49d5-afbf-91cd9cfcf130">

5. go to the http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions and verify that the button link includes the `utm_source` and `utm_campaign`
<img width="1631" alt="image" src="https://github.com/user-attachments/assets/a4e0cede-9619-4e6c-9129-76fe860cc61a">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Add UTM parameters to missing payment method notice links.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
